frozendict/frozendict.pyi

```diff
@@ -1,6 +1,57 @@
-try:
-    from ._frozendict import *
-except ImportError:
-    from .core import *
+import collections
+from _typeshed import Self
+from collections.abc import Iterable, Iterator, Mapping
+from typing import Any, Generic, NoReturn, Tuple, TypeVar, Union, overload
 
-FrozenOrderedDict = frozendict
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+def frozendict_or(
+    self: Mapping[_KT, _VT], other: Mapping[_KT, _VT]
+) -> "frozendict[_KT, _VT]": ...
+
+class frozendict(Mapping[_KT, _VT], Generic[_KT, _VT]):
+    # Fake __init__ to describe what __new__ does:
+    dict_cls: type[dict[Any, Any]] = ...
+    @overload
+    def __init__(self, **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, mapping: Mapping[_KT, _VT]) -> None: ...
+    @overload
+    def __init__(self, iterable: Iterable[tuple[_KT, _VT]]) -> None: ...
+
+    # Magic Methods:
+    def __getitem__(self, __key: _KT) -> _VT: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[_KT]: ...
+    def __hash__(self) -> int: ...
+    def __repr__(self) -> str: ...
+    def copy(self: Self) -> Self: ...
+    def __copy__(self: Self) -> Self: ...
+    def __deepcopy__(self: Self) -> Self: ...
+    # Omit __reduce__, its used for Pickle and we don't need the annotation in code.
+    def set(self, key: _KT, value: _VT) -> frozendict[_KT, _VT]: ...
+    # This weird type annotation is to support subclassing frozendict. It should
+    # resolve to `frozendict[_KT, _VT]`.
+    def setdefault(self: Self, key: _KT, default: _VT) -> Union[Self, frozendict[_KT, _VT]]: ...
```

Comment:
Not sure to understand. Furthermore, why only here?

Reply:
This is in case someone subclasses frozendict (e.g. as an OrderedFrozenDict).
Do you want me to support OrderedFrozenDict?

Reply:
`dict` is ordered by default from Python 3.6. `frozendict` code is based on `dict` code. Notice furthermore that Python 3.6 is EOL and this package supports only Python 3.6+